### ISSUE TITLE
errors for valid imports [HuggingFace]

### ISSUE THREAD

**Issue by danjenson** [microsoft/pyright#3144]
**Describe the bug**
Import errors for valid imports.

**To Reproduce**
Import anything from hugging face `transformers` library.

**Expected behavior**
No import errors for working imports.

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/4793438/156717714-7fc942ce-f5d4-4c0c-8bbb-749735a5e78d.png)

**VS Code extension or command-line**
pyright with neovim

**Comment by erictraut** [comment]
These errors are correct. This library is a "py.typed" library, so pyright uses [these rules](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface) to determine which symbols are re-exported from each module in the library. 

For example, `transformers/__init__.py` contains the following statement:
```python
from .trainer import Trainer
```

If the library intends to re-export `Trainer` from this module, it needs to use a redundant form of the import statement, like this:

```python
from .trainer import Trainer as Trainer
```

If you think that it is the intent of this library to re-export these symbols, please consider filing a bug report or a PR in the corresponding repo.

You can tell pyright to ignore this condition by disabling the `reportPrivateImportUsage` diagnostic rule.
